pkg/crc/cluster/cluster.go

```diff
@@ -119,20 +120,82 @@ func UpdateClusterID(ocConfig oc.Config) error {
 	return nil
 }
 
-func AddProxyConfigToCluster(ocConfig oc.Config, proxy *network.ProxyConfig) error {
-	cmdArgs := []string{"patch", "proxy", "cluster", "-p",
-		fmt.Sprintf(`'{"spec":{"httpProxy":"%s", "httpsProxy":"%s", "noProxy":"%s"}}'`, proxy.HTTPProxy, proxy.HTTPSProxy, proxy.GetNoProxyString()),
-		"-n", "openshift-config", "--type", "merge"}
+func AddProxyConfigToCluster(sshRunner *ssh.Runner, ocConfig oc.Config, proxy *network.ProxyConfig) error {
+	type trustedCA struct {
+		Name string `json:"name"`
+	}
+
+	type proxyPatch struct {
+		HTTPProxy  string    `json:"httpProxy"`
+		HTTPSProxy string    `json:"httpsProxy"`
+		NoProxy    string    `json:"noProxy"`
+		TrustedCA  trustedCA `json:"trustedCA"`
+	}
+
+	type spec struct {
```

Comment:
... and this `patchSpec` if you want to have the name `patch` in one of the types.

Reply:
yes `patchSpec` make more sense :+1: